### PR TITLE
feat(rpc-types): Replace u8 with Connectedness Enum

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
Replaced u8 types with the Connectedness enum 

ref of the implementation https://github.com/libp2p/go-libp2p/blob/d55bed5f78fad923353b92c7d1db64c758bcbf9c/core/network/network.go#L49

related to https://github.com/ethereum-optimism/optimism/blob/40750a58e7a4a6f06370d18dfe6c6eab309012d9/op-node/p2p/rpc_api.go#L25

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
this makes sense, but this should ensure that this is serialized/deserialized as u8, which default serde derive doesn't do afaik.

so I'd also like to see a test for this

File: crates/rpc-types/src/net.rs [hunk 1]
```diff
@@ -100,3 +100,54 @@ pub struct PeerStats {
     pub banned: u32,
     pub known: u32,
 }
+
+/// Represents the connectivity state of a peer in a network, indicating the reachability and
+/// interaction status of a node with its peers.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Copy)]
+#[serde(rename_all = "camelCase")]
```
Comment (refcell) [inline crates/rpc-types/src/net.rs:107]: We'll want to add the u8 representation. Also should probably mark `NotConnected` as the `#[default]`

```rust
#[repr(u8)]
```

File: crates/rpc-types/src/net.rs [hunk 2]
```diff
@@ -100,3 +100,160 @@ pub struct PeerStats {
...
+
+impl<'de> Deserialize<'de> for Connectedness {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = u8::deserialize(deserializer)?;
+        Ok(Connectedness::from(value))
+    }
+}
```
Comment (refcell) [inline crates/rpc-types/src/net.rs:173]: We should be able to remove these and place `#[derive(Serialize, Deserialize)]` above `Connectedness`.

Comment (DoTheBestToGetTheBest) [reply crates/rpc-types/src/net.rs:173]: > We should be able to remove these and place `#[derive(Serialize, Deserialize)]` above `Connectedness`.

Yes but the tests are failling if i derive the Serialize and Deserialize

Comment (refcell) [reply crates/rpc-types/src/net.rs:173]: Let me reproduce and check it out

**Review by refcell — APPROVED** [review]

**Pull request merged**